include/dsl/Archive.hh

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+namespace dsl {
+   class DataStore;
+
+   class Archive {
+      public:
+         Archive(std::string const&, dsl::DataStore*, long long, long long, bool, long long);
```

Comment:
would it be possible to use exact size ints, like int64_t, uint32_t?

Reply:
This was a copy-paste from `nm`. The 3rd and 4th arguments are very likely `dsl::idstring` (classes with their first (and only?) variable `uint64_t`, hence they are typedef'd in this PR).

Reply:
Okay, I see :)

Reply:
@ZNixian i would prefer inlining the type definitions so that standard integer sizes are used at the point of declaration.

Reply:
I'm not quite sure what you mean - do you want the `long long`s swapped out for `void*` or `uint64_t`, depending on their purpose?

Reply:
Yes, I would prefer uint64_t. If you know it might be a pointer you could do that too, but they are fundamentally the same.

Reply:
I'll do that now.

Reply:
Done in 13bc8a6